pkg/prometheus/rules.go

```diff
@@ -190,32 +189,43 @@ func (c *Operator) selectRules(p *monitoringv1.Prometheus, namespaces []string)
 	)
 
 	for _, ns := range namespaces {
-		var marshalErr error
 		err := c.ruleInfs.ListAllByNamespace(ns, ruleSelector, func(obj interface{}) {
 			promRule := obj.(*monitoringv1.PrometheusRule).DeepCopy()
 			if err := k8sutil.AddTypeInformationToObject(promRule); err != nil {
 				level.Error(c.logger).Log("msg", "failed to set rule type information", "namespace", ns, "err", err)
 				return
 			}
 
-			if err := nsLabeler.EnforceNamespaceLabel(promRule); err != nil {
-				marshalErr = err
-				return
-			}
-
-			content, err := generateRulesConfiguration(promRule.Spec, c.logger)
-			if err != nil {
-				marshalErr = err
-				return
-			}
-			rules[fmt.Sprintf("%v-%v-%v.yaml", promRule.Namespace, promRule.Name, promRule.UID)] = content
+			promRules[fmt.Sprintf("%v-%v-%v.yaml", promRule.Namespace, promRule.Name, promRule.UID)] = promRule
 		})
 		if err != nil {
-			return nil, err
+			return nil, errors.Wrapf(err, "failed to list prometheus rules in namespace %s", ns)
 		}
-		if marshalErr != nil {
-			return nil, marshalErr
+	}
+
+	var rejected int
+	rules := make(map[string]string, len(promRules))
+
+	for ruleName, promRule := range promRules {
+		var err error
+		if err := nsLabeler.EnforceNamespaceLabel(promRule); err != nil {
+			continue
+		}
+
+		content, err := generateRulesConfiguration(promRule.Spec, c.logger)
+		if err != nil {
+			rejected++
+			level.Warn(c.logger).Log(
+				"msg", "skipping prometheusrule",
+				"error", err.Error(),
+				"prometheusrule", promRule.Name,
+				"namespace", p.Namespace,
```

Comment:
shouldn't it be the rule namespace? Since users will want to know in which namespace is the prometheus rule that contains the error 

```suggestion
				"namespace", promRule.Namespace,
```

Reply:
Correct. my bad :)